src/main/java/com/fulcrumgenomics/util/DemuxFastqs.java

```diff
@@ -677,14 +697,24 @@ void convertQuality(final byte[] quals, final FastqQualityFormat version) {
             return sampleBarcodeQualities;
         }
 
-        private String getMolecularBarcode(final List<FastqRecord>  records) {
-            final byte[][] molecularBarcode = new byte[records.size()][];
+        private byte[][] getMolecularBarcodes(final List<FastqRecord> records) {
+            final byte[][] molecularBarcodes = new byte[records.size()][];
+            int numNonEmptyMolecularBarcodes = 0;
             for (int i = 0; i < records.size(); i++) {
                 final FastqRecord record = records.get(i);
                 final ReadStructureInfo readStructureInfo = readStructureInfos.get(i);
-                molecularBarcode[i] = getBasesAtCycles(record.getReadString().getBytes(), readStructureInfo.molecularBarcodeCycles);
+                molecularBarcodes[i] = getBasesAtCycles(record.getReadString().getBytes(), readStructureInfo.molecularBarcodeCycles);
```

Comment:
If `8M8B8M76T` is the read structure for one of index1/2 or read1/2, then yes.  Unlike `IlluminaBasecallsToSam` and friends, the read structure is per read, not across all cycles.